src/algoliasearch.helper.js

```diff
@@ -150,7 +150,7 @@ AlgoliaSearchHelper.prototype.searchOnce = function(options, cb) {
     return this.client.search(
       queries,
       function(err, content) {
-        cb(err, new SearchResults(tempState, content), tempState);
+        cb(err, !err ? new SearchResults(tempState, content) : null, tempState);
```

Comment:
Good catch! Could you use an explicit if on the line before? Maybe use if / else to show which case is which and dumb down the paramters? Something like:

```js
if(err) cb(err, null, tempState);
else cb(null, SearchResults(tempState, content), tempState);
```